docstrings as per #348, backend=numpy, additional state variables

Peter230655 opened the pull request:
- using ``backend='numpy' `` reduced running time on my PC from anout 13 sec to about 7 sec.
- storing the solution in .csv, the accuracy of ``fmt='%.2f' `` may be too inaccurate. Here I needed ``fmt='%.12f' ``
- added additional state variables to make it more realistic: at the final point, not only the speed of the load, but also its acceleration must be zero. So, I use $h_i = \dfrac{d^2}{dt^2}(\textrm{position}_i \textrm{of load})$ and set $h_i(t_f) = 0$

moorepants commented on examples-gallery/intermediate/plot_crane_moving_a_load.py:
```diff
@@ -1,14 +1,33 @@
+----------
+
+- Show the use of ``opty's variable node time interval`` feature to solve a
+  relatively simple problem.
```
![image](https://github.com/user-attachments/assets/9c5021fb-7940-4422-b9bc-9215a2121e57)

remove monospaced phrase

Peter230655 replied:
You then remove the double backwards strokes?

moorepants replied:
yes

Peter230655 replied:
> yes

done

moorepants commented:
No thumbnail is showing:

Peter230655 commented:
> No thumbnail is showing: ![image](https://private-user-images.githubusercontent.com/276007/416987109-375e7344-c05d-4296-b7f9-a6817348b068.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1NDgzMDYsIm5iZiI6MTc0MDU0ODAwNiwicGF0aCI6Ii8yNzYwMDcvNDE2OTg3MTA5LTM3NWU3MzQ0LWMwNWQtNDI5Ni1iN2Y5LWE2ODE3MzQ4YjA2OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyNlQwNTMzMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01N2FiOTk5MWZiYWE4YWFlYjYxNjFkOTAxODM5MjljMGZiODAwMzE5YmE4ZWFkYmM2OTRhYTk3NGRiYjZhODMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._yfe0_0iN1HR5_gs7SzCrLJmJmQb3pjbQcj3kvU4-Eo)

Strange: when I do ...make html it shows a thumbnail
![image](https://github.com/user-attachments/assets/67b28fda-7bc1-4f97-bc62-9247d26cacb0)

moorepants commented:
make html locally depends on the cached prior runs of make html. So it may not reflect a clean build.

Peter230655 commented:
> make html locally depends on the cached prior runs of make html. So it may not reflect a clean build.

Would this even be so if I turn off my PC and turn it back on?
Would it use cached version on a simulation even if I changed something in that simulation?
(I get the feel, if I changed something, it runs again...)

moorepants commented:
> Would this even be so if I turn off my PC and turn it back on?

No, that's not how it works.

When you run `make html` it creates a new directory: `opty/docs/examples'. You have to delete that directory to "clear the cache". Also running `make clean` deletes all other sphinx generated output. After both those things you can run `make html` again for a fresh build (will take time).

Peter230655 commented:
> The thumbnail does not show on readthedocs, so it has to be figured out.

But how can I try things, if it shows with ``make html`` ?

moorepants commented:
You can try lots of things, but I don't know the answer to every single reason things don't work. I search the internet and try things until enough clues come together to figure out what is going on. All I know at the moment is that the thumbnail does not display correctly when doing a clean build on read the docs and that is the mechanism we publish the official documentation website. Why that is different on your local computer, I don't know. Mostly because I don't have 100% information about what happens on your computer.

moorepants commented:
> My problem is: how can I test the effect of a change I make? I cannot push every change for you to see what has happened!

Yes, you have to learn to debug on your own :)

>  Can I do a build on read the docs without pushing it?

No, it is a cloud service we use and you can only look at the error logs it produces.

>  As regards searching the internet, I would not know what to search for.

I don't know either, but I try until something works.

moorepants commented:
> So a simulation may work with opty/docs make html, it may pass your 13 tests - and still not work right?

Yes.

moorepants commented:
```
computation time summary:
    - ../examples-gallery/advanced/plot_car_around_pylons.py:                             89.45 sec   0.0 MB
    - ../examples-gallery/advanced/plot_ball_rolling_on_spinning_disc.py:                 71.70 sec   0.0 MB
    - ../examples-gallery/advanced/plot_park2004.py:                                      70.09 sec   0.0 MB
    - ../examples-gallery/advanced/plot_sit_to_stand.py:                                  69.38 sec   0.0 MB
    - ../examples-gallery/intermediate/plot_drone.py:                                     67.70 sec   0.0 MB
    - ../examples-gallery/intermediate/plot_parallel_park.py:                             63.32 sec   0.0 MB
    - ../examples-gallery/beginner/plot_sliding_block.py:                                 61.46 sec   0.0 MB
    - ../examples-gallery/beginner/plot_pendulum_swing_up_fixed_duration.py:              59.10 sec   0.0 MB
    - ../examples-gallery/beginner/plot_pendulum_swing_up_variable_duration.py:           55.38 sec   0.0 MB
    - ../examples-gallery/advanced/plot_one_legged_time_trial.py:                         45.35 sec   0.0 MB
    - ../examples-gallery/intermediate/plot_mississippi_steamboat.py:                     44.22 sec   0.0 MB
    - ../examples-gallery/advanced/plot_human_gait.py:                                    41.11 sec   0.0 MB
    - ../examples-gallery/advanced/plot_particle_in_tube.py:                              40.50 sec   0.0 MB
    - ../examples-gallery/advanced/plot_car_in_garage.py:                                 31.28 sec   0.0 MB
    - ../examples-gallery/intermediate/plot_countersteer.py:                              28.54 sec   0.0 MB
    - ../examples-gallery/beginner/plot_betts_10_103_104.py:                              21.56 sec   0.0 MB
    - ../examples-gallery/beginner/plot_betts_10_7.py:                                    20.48 sec   0.0 MB
    - ../examples-gallery/beginner/plot_betts_10_50.py:                                   13.94 sec   0.0 MB
    - ../examples-gallery/intermediate/plot_non_contiguous_parameter_identification.py:   10.77 sec   0.0 MB
    - ../examples-gallery/beginner/plot_vyasarayani.py:                                   10.24 sec   0.0 MB
    - ../examples-gallery/intermediate/plot_crane_moving_a_load.py:                       10.22 sec   0.0 MB
    - ../examples-gallery/beginner/plot_betts2003.py:                                      9.84 sec   0.0 MB
    - ../examples-gallery/intermediate/plot_two_link_pendulum_on_a_cart.py:                9.70 sec   0.0 MB
    - ../examples-gallery/advanced/model_park2004.py:                                      0.00 sec   0.0 MB
  ```

Peter230655 commented:
> ```
> computation time summary:
>     - ../examples-gallery/advanced/plot_car_around_pylons.py:                             89.45 sec   0.0 MB
>     - ../examples-gallery/advanced/plot_ball_rolling_on_spinning_disc.py:                 71.70 sec   0.0 MB
>     - ../examples-gallery/advanced/plot_park2004.py:                                      70.09 sec   0.0 MB
>     - ../examples-gallery/advanced/plot_sit_to_stand.py:                                  69.38 sec   0.0 MB
>     - ../examples-gallery/intermediate/plot_drone.py:                                     67.70 sec   0.0 MB
>     - ../examples-gallery/intermediate/plot_parallel_park.py:                             63.32 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_sliding_block.py:                                 61.46 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_pendulum_swing_up_fixed_duration.py:              59.10 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_pendulum_swing_up_variable_duration.py:           55.38 sec   0.0 MB
>     - ../examples-gallery/advanced/plot_one_legged_time_trial.py:                         45.35 sec   0.0 MB
>     - ../examples-gallery/intermediate/plot_mississippi_steamboat.py:                     44.22 sec   0.0 MB
>     - ../examples-gallery/advanced/plot_human_gait.py:                                    41.11 sec   0.0 MB
>     - ../examples-gallery/advanced/plot_particle_in_tube.py:                              40.50 sec   0.0 MB
>     - ../examples-gallery/advanced/plot_car_in_garage.py:                                 31.28 sec   0.0 MB
>     - ../examples-gallery/intermediate/plot_countersteer.py:                              28.54 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_betts_10_103_104.py:                              21.56 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_betts_10_7.py:                                    20.48 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_betts_10_50.py:                                   13.94 sec   0.0 MB
>     - ../examples-gallery/intermediate/plot_non_contiguous_parameter_identification.py:   10.77 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_vyasarayani.py:                                   10.24 sec   0.0 MB
>     - ../examples-gallery/intermediate/plot_crane_moving_a_load.py:                       10.22 sec   0.0 MB
>     - ../examples-gallery/beginner/plot_betts2003.py:                                      9.84 sec   0.0 MB
>     - ../examples-gallery/intermediate/plot_two_link_pendulum_on_a_cart.py:                9.70 sec   0.0 MB
>     - ../examples-gallery/advanced/model_park2004.py:                                      0.00 sec   0.0 MB
> ```

Looks line using the solution to skip solve(...) and backend='numpy' has seriously reduced running times.

Peter230655 commented:
> > ```
> > computation time summary:
> >     - ../examples-gallery/advanced/plot_car_around_pylons.py:                             89.45 sec   0.0 MB
> >     - ../examples-gallery/advanced/plot_ball_rolling_on_spinning_disc.py:                 71.70 sec   0.0 MB
> >     - ../examples-gallery/advanced/plot_park2004.py:                                      70.09 sec   0.0 MB
> >     - ../examples-gallery/advanced/plot_sit_to_stand.py:                                  69.38 sec   0.0 MB
> >     - ../examples-gallery/intermediate/plot_drone.py:                                     67.70 sec   0.0 MB
> >     - ../examples-gallery/intermediate/plot_parallel_park.py:                             63.32 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_sliding_block.py:                                 61.46 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_pendulum_swing_up_fixed_duration.py:              59.10 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_pendulum_swing_up_variable_duration.py:           55.38 sec   0.0 MB
> >     - ../examples-gallery/advanced/plot_one_legged_time_trial.py:                         45.35 sec   0.0 MB
> >     - ../examples-gallery/intermediate/plot_mississippi_steamboat.py:                     44.22 sec   0.0 MB
> >     - ../examples-gallery/advanced/plot_human_gait.py:                                    41.11 sec   0.0 MB
> >     - ../examples-gallery/advanced/plot_particle_in_tube.py:                              40.50 sec   0.0 MB
> >     - ../examples-gallery/advanced/plot_car_in_garage.py:                                 31.28 sec   0.0 MB
> >     - ../examples-gallery/intermediate/plot_countersteer.py:                              28.54 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_betts_10_103_104.py:                              21.56 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_betts_10_7.py:                                    20.48 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_betts_10_50.py:                                   13.94 sec   0.0 MB
> >     - ../examples-gallery/intermediate/plot_non_contiguous_parameter_identification.py:   10.77 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_vyasarayani.py:                                   10.24 sec   0.0 MB
> >     - ../examples-gallery/intermediate/plot_crane_moving_a_load.py:                       10.22 sec   0.0 MB
> >     - ../examples-gallery/beginner/plot_betts2003.py:                                      9.84 sec   0.0 MB
> >     - ../examples-gallery/intermediate/plot_two_link_pendulum_on_a_cart.py:                9.70 sec   0.0 MB
> >     - ../examples-gallery/advanced/model_park2004.py:                                      0.00 sec   0.0 MB
> > ```
> 
> Looks line using the solution to skip solve(...) and backend='numpy' has seriously reduced running times.
